javascript/javascript3/week2/promise.js

```diff
@@ -0,0 +1,93 @@
+ // Promise that resolves after set time
+
+function getResolveIn(resolveAfter) {
+   return new Promise(function(resolve, reject){
+       setTimeout(function() {
+           resolve(resolveAfter);
+
+       }, 1000 * resolveAfter);
+   });
+}
+getResolveIn(8).then(function() {
+   console.log("I am called asynchronously")
+});
+
+
+// Using the above promise with async/await
+
+async function getResolveIn(resolveAfter) {
+    let promise = new Promise((resolve, reject) => {
+        setTimeout(() => resolve("Calling after 8 seconds"), 1000 * resolveAfter)
+    });
+    let result = await promise;          // wait until the promise resolves.
+    console.log(result);
+}
+getResolveIn(4);
```

Comment:
You wrote the "getResolveIn" above and the code is reusable. You just need to call it.

async function getResolveInAsync(resolveAfter) {
    await getResolveIn(resolveAfter); 
    console.log("Async await is executed");
}

getResolveInAsync(4);